File: backend/src/modules/configuracion/dto/configurar-tienda.dto.ts
import { IsOptional, IsString, IsNumber, IsBoolean, Min } from 'class-validator';

export class ConfigurarTiendaDto {
  @IsOptional()
  @IsString()
  nombreTienda?: string;

  @IsOptional()
  @IsString()
  whatsappContacto?: string;

  @IsOptional()
  @IsString()
  banco?: string;

  @IsOptional()
  @IsString()
  cbu?: string;

  @IsOptional()
  @IsString()
  alias?: string;

  @IsOptional()
  @IsString()
  titular?: string;

  @IsOptional()
  @IsString()
  mensajeTransferencia?: string;

  @IsOptional()
  @IsNumber()
  @Min(1)
  pedidoVencimientoHoras?: number;

  @IsOptional()
  @IsBoolean()
  estadoProductoBorrador?: boolean;
}

File: backend/src/main/java/com/sistemapedidos/modules/configuracion/dto/ConfigurarTiendaDto.java
package com.sistemapedidos.modules.configuracion.dto;

import jakarta.validation.constraints.Min;

public record ConfigurarTiendaDto(
        String nombreTienda,
        String whatsappContacto,
        String banco,
        String cbu,
        String alias,
        String titular,
        String mensajeTransferencia,
        @Min(value = 1, message = "Las horas de vencimiento deben ser al menos 1") Integer pedidoVencimientoHoras,
        Boolean estadoProductoBorrador) {
}
